src/allmydata/scripts/tahoe_run.py

```diff
@@ -212,10 +212,11 @@ def run(config):
     if not nodetype:
         print("%s is not a recognizable node directory" % quoted_basedir, file=err)
         return 1
-    # Now prepare to turn into a twistd process. This os.chdir is the point
-    # of no return.
-    os.chdir(basedir)
-    twistd_args = ["--nodaemon"]
+
+    twistd_args = ["--nodaemon", "--rundir", basedir]
+    if sys.platform != "win32":
+        pidfile = get_pidfile(basedir)
+        twistd_args.extend(["--pidfile", pidfile])
```

Comment:
Where did the PID file come from before if this wasn't set?

Reply:
The code used to rely on the default pidfile behavior from twistd.  That implicit behavior depended on interactions with `os.chdir` to get a predictable path which made the tests more complicated (eg they previously monkey-patched `os.chdir` to keep some kind of control over what was happening).